### ISSUE TITLE
Regression on installing multiple editable dependencies

### ISSUE THREAD

**Issue by hsheth2** [astral-sh/uv#1787]
Installing multiple editable versions causes a panic. This used to work with uv 0.1.5, and works with `pip`.

```shell
$ uv --version
uv 0.1.6
$ uname -mo
Darwin x86_64
$ RUST_BACKTRACE=1 uv pip install -e '/Users/hsheth/projects/datahub/metadata-ingestion/' -e '/Users/hsheth/projects/datahub/metadata-ingestion-modules/airflow-plugin/[plugin-v2]'
   Built file:///Users/hsheth/projects/datahub/metadata-ingestion
   Built file:///Users/hsheth/projects/datahub/metadata-ingestion-modules/airflow-plugin                                                                                              
Built 2 editables in 4.54s
⠹ pycparser==2.21                                                                                                                                                                     
thread 'main' panicked at crates/uv-resolver/src/resolution.rs:125:37:
Every package should be pinned: PackageName("acryl-datahub-airflow-plugin")
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

See https://github.com/datahub-project/datahub/pull/9885 for a live example, or you can also repro by cloning the https://github.com/datahub-project/datahub repo and installing these two paths.

**Comment by charliermarsh** [comment]
I believe I know the problem — thanks.

**Comment by charliermarsh** [comment]
Is there any relationship between the two? Do they depend on one another? Are there any extras involved?

**Comment by hsheth2** [comment]
The base `acryl-datahub-airflow-plugin` package depends on `acryl-datahub`. The extra is only adding a few other dependencies, not doesn't impact the relationship between the two editable packages.

**Comment by charliermarsh** [comment]
Perfect, thanks.

**Comment by charliermarsh** [comment]
I’ll make sure it’s fixed by the next release.

**Comment by charliermarsh** [comment]
Ugh, sadly the comment you pasted above works for me on `main` when run on that repo.

**Comment by charliermarsh** [comment]
I'm trying to reproduce. What exact commit did you run this against, in the datahub repo?

**Comment by hsheth2** [comment]
The CI failure is here https://github.com/datahub-project/datahub/actions/runs/7981141634/job/21792271496?pr=9885. Because we're using the default checkout action, CI ran on a commit that merged that PR into master

That said, I am trying the same commands again locally and it also seems to work for me. I might need to keep digging here

**Comment by hsheth2** [comment]
Seems like the issue shows up when using `-e` in conjunction with an extra that doesn't exist. I don't think the fact that there's multiple `-e` directives is related.

I've been able to work around it for now by declaring the `ignore` extra.

**Comment by charliermarsh** [comment]
Thanks -- I'll try to get back to this today.

**Comment by charliermarsh** [comment]
Thanks, this was my mistake. Just a bug.

**Comment by charliermarsh** [comment]
(The bug was in the error reporting path for editable installs with non-existent extras.)

**Comment by charliermarsh** [comment]
Should be fixed in v0.1.7 (out now).
